superset/db_engine_specs/base.py

```diff
@@ -1594,6 +1598,17 @@ def get_impersonation_key(cls, user: Optional[User]) -> Any:
         """
         return user.username if user else None
 
+    @classmethod
+    def get_public_information(cls) -> Dict[str, Any]:
+        """
+        Construct a Dict with properties we want to expose.
+
+        :returns: Dict with properties of our class like allows_file_upload
+        """
+        return {
+            "allows_file_upload": cls.allows_file_upload,
```

Comment:
how is this different from the current colum `allow_file_upload`? If they are very different, then should we think about changing the name of this? 

`allows_file_upload` and `allow_file_upload` being different columns that serve different purposes is going to be way too confusing.

Reply:
The `allow_file_upload` is in the `Database` model and it's set per each user's DB. While the `allows_file_upload` is set in the `DB Engine Spec` and cannot be set by the user. So, even when a `Database` has its flag set to true (think of existing user's DB), if its `DB Engine Spec` doesn't support it, we prevent file uploading to it.

Reply:
ok that makes sense! 

Could we name it `supports_file_upload` or something along the like? Having two columns only separated by a singular/plural shift is going to be confusing.

Reply:
Yeah, totally! Let me make the change and ILYK. Thanks for the input @AAfghahi

Reply:
of course! Otherwise this LGTM!

Reply:
PR Updated with the new `supports_file_upload` @AAfghahi  :)